### PR TITLE
test(renderer): App-level integration coverage for memo, cut/paste, review demotion, overlap

### PR DESCRIPTION
**Pull request by melly-lgtm**
## What

Four App.tsx integration gaps the matrix buildout's coverage critic flagged — wiring the per-module unit tests don't exercise. **No src changes; 13 tests, full suite green at 96.42% coverage.**

- **App.memoWiring** (5): the ComposerPopover audience switch end-to-end. `(doc, memo)` and `(span, memo)` store `comment.agent:false` and emit a `comment_created` payload with `agent:false`; the default *talk-to-agent* stores no `agent` field; a doc memo differs from a normal doc comment **solely** by `{agent:false}`. This is what `docForAgent` keys off to strip memos from the agent's view.
- **App.cutPaste.matrix** (4): `onCutComments`/`onPasteComments` over `clipboard.ts`. Cut splices the anchored thread out of both rail and body (`comment_deleted` count 2); paste re-mints a colliding id distinct from every live id; pasting twice yields two distinct ids; a no-anchor cut/paste logs count 0.
- **App.reviewDemotion** (1): a span comment authored **during** review is demoted to `anchor:doc` on Apply (its anchor doesn't survive the base→agent diff), with its reply kept parent-linked; the saved doc passes `checkIntegrity` (no `dangling_link`/`span_missing_link`). Confirms demotion repairs replies — *this disproves a suspected latent bug*.
- **App.overlapGate** (3): `addComment`'s `spanCommentBlocker` guard fires when the body is rewritten under an open composer — overlap → `cantOverlap`, not-found → `cantAnchor`, clean → anchored.

## Test
`npx vitest run` (pre-commit gate) → 663 passing / 2 skipped, statements/lines 96.42% (≥95%).

<a href="https://cubic.dev/pr/melly-lgtm/inplan/pull/14?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Tests**
  * Added integration test coverage for span-comment cut/paste threading, including ID reminting, reply carrying, and clipboard edge cases.
  * Added tests for comment composer audience (“memo” vs default talk-to-agent), verifying stored payloads and anchor differences for doc vs span comments.
  * Added tests enforcing span-comment anchoring rules, including overlap prevention and selection-anchoring failures.
  * Added a review workflow test to confirm span-comment demotion to doc-level after proposal acceptance while preserving thread/replies and integrity checks.

### CODE REVIEW

File: packages/renderer/test/App.memoWiring.test.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+/** Hand off the turn so the host performs a canonical save — a SPAN comment changes the body
+ *  (its `[text](#cmt-id)` anchor link), which in Turn mode isn't persisted until the next save.
+ *  (A doc-level comment is comment-only and already persists immediately.) */
+async function flushTurn() {
+  await act(async () => {
+    screen.getByRole("button", { name: /finish turn/i }).click();
+  });
+}
```
[the remark this answers is not in the document]
Comment (melly-lgtm) [reply packages/renderer/test/App.memoWiring.test.tsx:68]: Fixed in 25c694f — `flushTurn()` now awaits a persistence barrier (`waitFor` a parsed comment in `api.load()`) after clicking Finish turn, so `storedComment()` reads the canonical saved state instead of racing the fire-and-forget save.

**Comment by melly-lgtm** [comment]
Addressed both CodeRabbit findings in 25c694f:
- **App.memoWiring** (post-turn race): `flushTurn()` now awaits the async canonical save before `storedComment()` reads `api.load()`.
- **App.overlapGate** (ambiguous `waitFor`, summary nitpick): the external-rewrite step now syncs on the rewrite-specific anchor `[data-cmt="cmt-abc123"]` (present only after the rewrite) instead of `"here."` (which appears in both the original and rewritten body).

Both files pass; pre-commit coverage gate held.
